ci/src/revisions/git_lib.py

```diff
@@ -0,0 +1,163 @@
+import argparse
+import logging
+import pathlib
+import subprocess
+import sys
+from typing import List
+
+
+class HelpfulParser(argparse.ArgumentParser):
+    """An argparse parser that prints usage on any error."""
+
+    def error(self, message):
+        sys.stderr.write("error: %s\n" % message)
+        self.print_help()
+        sys.exit(2)
+
+
+def init_helpful_parser() -> HelpfulParser:
+    parser = HelpfulParser()
+    parser.add_argument("--verbose", "-v", action="store_true", help="Verbose mode")
+
+    return parser
```

Comment:
why not include all args here?

Reply:
Since this is a library. At the end it does have a `main` but I wanted this to be extensible yet still follow the same way of creating the parser.

Reply:
Doesn't sounds like a library then. A library should not be called as an executable. Seems like we should give it another go with refactoring, I think this can be improved.

Reply:
I quite like the new code! Resolving!